orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -780,10 +841,19 @@ Status TraverseGraphWithConnectedElement(Graph& graph,
                                          std::set<NodeArg*>& visited_inputs,
                                          std::set<NodeArg*>& visited_outputs) {
   assert(start_node);
+
   visited_nodes.clear();
   visited_inputs.clear();
   visited_outputs.clear();
 
+  for (const auto node_arg : graph.GetInputs()) {
```

Comment:
`GetInputs()` returns `const std::vector<const NodeArg*>&`. Which pointer do you want to avoid?


---
In reply to: [527638153](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r527638153) [](ancestors = 527638153)